create-publication-from-doi/src/main/java/no/unit/nva/doi/fetch/MainHandler.java

```diff
@@ -0,0 +1,151 @@
+    public static final JsonPointer ORG_NUMBER = JsonPointer.compile("/authorizer/claims/custom:orgNumber");
+    public static final String NULL_DOI_URL_ERROR = "doiUrl can not be null";
+
+    private final ObjectMapper objectMapper;
+    private final transient PublicationConverter publicationConverter;
+    private final transient DoiTransformService doiTransformService;
+    private final transient DoiProxyService doiProxyService;
+    private final transient PublicationPersistenceService publicationPersistenceService;
+    private final transient String publicationApiHost;
+    private final transient String publicationApiScheme;
+
+    private static final Logger logger = LoggerFactory.getLogger(MainHandler.class);
+
+    @JacocoGenerated
+    public MainHandler() {
+        this(JsonUtils.objectMapper, new PublicationConverter(), new DoiTransformService(),
+            new DoiProxyService(), new PublicationPersistenceService(), new Environment());
+    }
+
+    /**
+     * Constructor for MainHandler.
+     *
+     * @param objectMapper objectMapper.
+     * @param environment  environment.
+     */
+    public MainHandler(ObjectMapper objectMapper,
+                       PublicationConverter publicationConverter,
+                       DoiTransformService doiTransformService,
+                       DoiProxyService doiProxyService,
+                       PublicationPersistenceService publicationPersistenceService,
+                       Environment environment) {
+        super(RequestBody.class, environment, logger);
+        this.objectMapper = objectMapper;
+        this.publicationConverter = publicationConverter;
+        this.doiTransformService = doiTransformService;
+        this.doiProxyService = doiProxyService;
+        this.publicationPersistenceService = publicationPersistenceService;
+        this.publicationApiHost = environment.readEnv(PUBLICATION_API_HOST_ENV);
+        this.publicationApiScheme = environment.readEnv(PUBLICATION_API_SCHEME_ENV);
+    }
+
+    @Override
+    protected Summary processInput(RequestBody input, RequestInfo requestInfo, Context context)
+        throws ApiGatewayException {
+
+        URI apiUrl = urlToPublicationProxy();
+        validate(input);
+
+        String owner = RequestUtils.getRequestContextParameter(requestInfo, FEIDE_ID);
+        String orgNumber = RequestUtils.getRequestContextParameter(requestInfo, ORG_NUMBER);
+        String authorization = RequestUtils.getHeader(requestInfo, HttpHeaders.AUTHORIZATION);
+
+        try {
+            Publication publication = getPublicationMetadata(input, owner, orgNumber);
+            PublicationResponse publicationResponse = tryInsertPublication(authorization, apiUrl, publication);
+            return publicationConverter
+                .toSummary(objectMapper.convertValue(publicationResponse, JsonNode.class));
+        } catch (URISyntaxException
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)